src/org/graphstream/algorithm/Dijkstra.java

```diff
@@ -509,9 +515,9 @@ public void remove() {
 		}
 	}
 
-	protected class TreeIterator<T extends Edge> implements Iterator<T> {
+	protected class TreeIterator implements Iterator<Edge> {
```

Comment:
For the record, java 9 has a `Stream.iterate` static method with a `next` and a `hasNext` function parameters, very similar to classical iterators.